app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixLocationFilterBottomSheet.kt

```diff
@@ -0,0 +1,319 @@
+                        }
+
+                        Row(
+                            modifier = Modifier.weight(0.15f).testTag("rightPriceDistText"),
+                            horizontalArrangement = Arrangement.Start) {
+                              Text(
+                                  text = "800 km",
+                                  color = colorScheme.onBackground,
+                                  style = TextStyle(fontWeight = FontWeight.SemiBold),
+                                  modifier = Modifier.testTag("rightDistTextValue"))
+                            }
+                      }
+
+                  Spacer(modifier = Modifier.height(verticalSpacing * 1.5f))
+
+                  Button(
+                      enabled = selectedOption.value != null,
+                      onClick = {
+                        if (selectedOption.value == 0) {
+                          if (locationHelper.checkPermissions()) {
+                            locationHelper.getCurrentLocation { loc ->
+                              loc?.let {
+                                val location = Location(it.latitude, it.longitude, "Phone Location")
+                                onApplyClick(location, range)
+                              }
+                            }
+                          } else {
+                            locationHelper.requestPermissions()
+                          }
+                        } else {
+                          onApplyClick(
+                              userProfile.locations[selectedOption.value!!.minus(1)], range)
+                        }
+                        onDismissRequest()
+                      },
+                      modifier =
+                          Modifier.fillMaxWidth()
+                              .padding(horizontal = buttonPaddingHorizontal)
+                              .testTag("applyButton"),
+                      colors =
+                          androidx.compose.material3.ButtonDefaults.buttonColors(
+                              containerColor = colorScheme.primary,
+                              contentColor = colorScheme.onPrimary)) {
+                        Text("Apply", modifier = Modifier.testTag("applyButtonText"))
+                      }
+                  Text(
+                      text = "Clear",
+                      color =
+                          if (clearEnabled) colorScheme.primary
+                          else colorScheme.onSecondaryContainer,
+                      modifier =
+                          Modifier.clickable(enabled = clearEnabled) {
+                                onClearClick()
+                                onDismissRequest()
+                              }
+                              .padding(vertical = verticalSpacing / 2)
+                              .testTag("resetButton"))
+                }
+          }
+        }
+  }
+}
+
+@Preview(showBackground = true)
+@Composable
+fun hehe() {
```

Comment:
Maybe rename this function with a name that is more meaningful.

Reply:
Will do.